### PR TITLE
[Formatter] Implement the missing node transform methods in NewFormattingTreeModifier and fix some failing formatter related tests

### PR DESCRIPTION
**Pull request by sanjana**
## Purpose

> - Implementation of `transform(ClassDefinitionNode classDefinitionNode)` and `transform(NamedWorkerDeclarator namedWorkerDeclarator)` methods
> - Fix some failing formatter related test cases

Fixes #26052

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 1]
```diff
@@ -3451,6 +3493,18 @@ public NamedWorkerDeclarationNode transform(NamedWorkerDeclarationNode namedWork
                 .apply();
     }
 
+    @Override
+    public NamedWorkerDeclarator transform(NamedWorkerDeclarator namedWorkerDeclarator) {
+        NodeList<StatementNode> workerInitStatements = formatNodeList(namedWorkerDeclarator.workerInitStatements(), 1
+                , 0, 1, 0);
+        NodeList<NamedWorkerDeclarationNode> namedWorkerDeclarations =
+                formatNodeList(namedWorkerDeclarator.namedWorkerDeclarations(), 0, 2, 0, 1);
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:3501]: This should be same as the statements within the block-statement.

File: misc/formatter/modules/formatter-core/src/test/resources/actions/type-cast/assert/type_cast_action_1.bal [hunk 2]
```diff
@@ -1,15 +1,9 @@
-type Person record {
-    string name;
-};
+type Person record { string name; };
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/actions/type-cast/assert/type_cast_action_1.bal:1]: I think we should expand module-level record type defs always.

I have fixed this with https://github.com/ballerina-platform/ballerina-lang/commit/51673806e91612de5f6aca3da9f0d59dadea07a2

**Review by SupunS — CHANGES_REQUESTED** [review]

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 3]
```diff
@@ -323,8 +323,13 @@ public FunctionDefinitionNode transform(FunctionDefinitionNode functionDefinitio
         Token functionKeyword = formatToken(functionDefinitionNode.functionKeyword(), 1, 0);
         IdentifierToken functionName = formatToken(functionDefinitionNode.functionName(), 0, 0);
         FunctionSignatureNode functionSignatureNode = formatNode(functionDefinitionNode.functionSignature(), 1, 0);
-        FunctionBodyNode functionBodyNode = formatNode(functionDefinitionNode.functionBody(),
-                this.trailingWS, this.trailingNL);
+        FunctionBodyNode functionBodyNode;
+        if (functionDefinitionNode.parent().kind() == SyntaxKind.CLASS_DEFINITION) {
```
[the remark this answers is not in the document]
Comment (sanjana) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:327]: But will it be possible to obtain this kind of formatting?
```
class Person {
    *A;
    public int age = 10;
    function foo() returns int {

        int a = 12345;
        return a;
    }

    public string name = "sample name";
    int year = 50;
    function bar() returns string {

        string b = "bar";
        return b;
    }

    string month = "february";
}
```

The following implementation also is using this approach as well. 
https://github.com/ballerina-platform/ballerina-lang/blob/51673806e91612de5f6aca3da9f0d59dadea07a2/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java#L536

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 4]
```diff
@@ -3413,8 +3409,35 @@ public OrderKeyNode transform(OrderKeyNode orderKeyNode) {
...
+        }
+
+        NodeList<Token> classTypeQualifiers = formatNodeList(classDefinitionNode.classTypeQualifiers(), 1, 0, 1, 0);
+        Token classKeyword = formatToken(classDefinitionNode.classKeyword(), 1, 0);
+        Token className = formatToken(classDefinitionNode.className(), 1, 0);
+        Token openBrace = formatToken(classDefinitionNode.openBrace(), 0, 1);
+        this.preserveNewlines = true;
+
+        indent();
+        NodeList<Node> members = formatNodeList(classDefinitionNode.members(), 0, 1, 0, 1);
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:3429]: Each member also should preserve newlines. So we need to pass the flag

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 5]
```diff
@@ -387,6 +387,7 @@ public RequiredParameterNode transform(RequiredParameterNode requiredParameterNo
     @Override
     public FunctionBodyBlockNode transform(FunctionBodyBlockNode functionBodyBlockNode) {
         Token openBrace = formatToken(functionBodyBlockNode.openBraceToken(), 0, 1);
+//        this.preserveNewlines = true;
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:390]: Lets remove this commented out line

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 6]
```diff
@@ -323,8 +323,8 @@ public FunctionDefinitionNode transform(FunctionDefinitionNode functionDefinitio
         Token functionKeyword = formatToken(functionDefinitionNode.functionKeyword(), 1, 0);
         IdentifierToken functionName = formatToken(functionDefinitionNode.functionName(), 0, 0);
         FunctionSignatureNode functionSignatureNode = formatNode(functionDefinitionNode.functionSignature(), 1, 0);
-        FunctionBodyNode functionBodyNode = formatNode(functionDefinitionNode.functionBody(),
-                this.trailingWS, this.trailingNL);
+        //TODO: Fix formatting issue when the function is within a class definition declaration.
+        FunctionBodyNode functionBodyNode = formatNode(functionDefinitionNode.functionBody(), this.trailingWS, this.trailingNL);
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:327]: This line might be longer than 120

File: misc/formatter/modules/formatter-core/src/test/resources/actions/query/assert/query_action_1.bal [hunk 7]
```diff
@@ -4,15 +4,13 @@ type Student record {
...
-        var name = {
-            firstName : student.firstName
+    Student[] studentList = [{
+        firstName: "Michelle",
+        gpa: 3.5
+    }];
+    var result = from var student in studentList
+        do {
+
+            var name = { firstName: student.firstName };
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/actions/query/assert/query_action_1.bal:14]: This assert is not correct. Let's keep the original for now.

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_13.bal [hunk 8]
```diff
@@ -1,5 +1,6 @@
 readonly distinct class Foo {
     string name;
+
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_13.bal:3]: This also looks wrong. Lets keep the original

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_16.bal [hunk 9]
```diff
@@ -2,7 +2,7 @@ class Student {
     public function foo() {
         string x = "abc";
     }
-
     int a;
     int b;
+
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_16.bal:7]: Same here

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_17.bal [hunk 10]
```diff
@@ -1,14 +1,8 @@
 class Foo {
     public function init() {
-
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_17.bal:3]: Same here

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_3.bal [hunk 11]
```diff
@@ -2,10 +2,8 @@ class Foo {
     public remote function foo() {
         int x = 0;
     }
-
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_3.bal:5]: Same

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_15.bal [hunk 12]
```diff
@@ -1,32 +1,32 @@
       client
    readonly
          distinct
-
-     class  Foo
+            class  Foo
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_15.bal:4]: same for this entire file

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_2.bal [hunk 13]
```diff
@@ -2,6 +2,8 @@ class
     Person
 
        {
+
+
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_2.bal:6]: same here

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java (hunk 5 above)
Comment (IrushiL) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:390]: Shall we remove this comment?

**Review by SupunS — CHANGES_REQUESTED** [review]

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java (hunk 6 above)
Comment (IrushiL) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:327]: This line seems to be exceeding the line limit.

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_13.bal (hunk 8 above)
Comment (sanjana) [reply misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_13.bal:3]: Isn't it a explicitly preserved newline added by the user?

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_3.bal (hunk 11 above)
Comment (sanjana) [reply misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/assert/class_definition_declaration_3.bal:5]: Could you please further elaborate?

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_15.bal (hunk 12 above)
Comment (sanjana) [reply misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_15.bal:4]: This is a source file, so should it be changed as well?

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_2.bal (hunk 13 above)
Comment (sanjana) [reply misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_2.bal:6]: This is also a source file. Should it be reformatted?

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_15.bal (hunk 12 above)
Comment (SupunS) [reply misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_15.bal]: ack

File: misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_2.bal (hunk 13 above)
Comment (SupunS) [reply misc/formatter/modules/formatter-core/src/test/resources/declarations/class-definition/source/class_definition_declaration_2.bal]: ack

**Review by SupunS — APPROVED** [review]

**Pull request merged**